src/lsm/level_data_iterator.zig

```diff
@@ -64,128 +70,101 @@ pub fn LevelDataIteratorType(comptime Table: type, comptime Storage: type) type
 
         pub fn init(
             allocator: mem.Allocator,
```

Comment:
Can we get rid of `allocator` here? It isn't used by `TableDataIterator` either, but there's a comment reserving it for future use 🤔.

https://github.com/tigerbeetledb/tigerbeetle/blob/8bb876834790e440e251626cc28a106a94692c8d/src/lsm/table_data_iterator.zig#L41

If so, we can also remove `deinit` for these structs.

Reply:
@chaitanyabhandari I think it has to do with read pipelining? But I'm not sure - so far I've only focused on the write path!

Reply:
Got it, thank you Fed! 

FYI, I've removed the unused allocator from `LevelIterator` & `TableDataIterator` as part of https://github.com/tigerbeetledb/tigerbeetle/pull/875/commits/376120a5eb12a8923eddb0d1d72e1f026c254b94! We can add it back if/when we require it for read pipelining.

cc: @batiati